napari/layers/image/image.py

```diff
@@ -1116,3 +1116,17 @@ def __init__(self, obj):
 
     def _raw_to_displayed(self, *args, **kwarg):
         return self.obj()._raw_to_displayed(*args, **kwarg)
+
+
+class AsyncImage(Image):
```

Comment:
@andy-sweet this is my example Layer class. I originally tried to implement one that inherited from Base, but there were too many required method implementations. This was the cleanest way.

Reply:
I think this seems fine. Though I'd suggest defining this class in the test module to make it clear this for testing only. That should also help you avoid some of the issues around defining new layer types.